[macOS WKTR] Enable async overflow scrolling by default for WPTs

aprotyas opened the pull request:
#### 6e655dfb81542a121845a4f664aaa0698bb307eb
<pre>
[macOS WKTR] Enable async overflow scrolling by default for WPTs
<a href="https://bugs.webkit.org/show_bug.cgi?id=263810">https://bugs.webkit.org/show_bug.cgi?id=263810</a>
<a href="https://rdar.apple.com/problem/117609234">rdar://problem/117609234</a>

Reviewed by Tim Nguyen.

This commit enables async overflow scrolling by default for WPTs on
macOS by hardcoding enablement of the `AsyncOverflowScrollingEnabled` web
preference within `TestControllerMac` when the test runner is running
WPTs.

The heuristic for determining whether the test running is a WPT is
simply the fact that the test path is a URL and is on the ports/hosts
specified in imported/w3c/resources/config.json, as prior art from
`shouldDumpJSConsoleLogInStdErr` shows.

We keep the concept of a valid WPT URL in a separate
`isWebPlatformTestURL` helper method in WPTFunctions.h that both
`shouldDumpJSConsoleLogInStdErr` and `shouldEnableAsyncOverflowScrolling`
can consult. To get all the definitions in the right place and satisfy
the linker, this commit also touches some of target membership for WKTR.

* LayoutTests/platform/mac-wk2/TestExpectations:
* Tools/TestRunnerShared/TestFeatures.cpp:
(WTR::shouldDumpJSConsoleLogInStdErr):
* Tools/TestRunnerShared/WPTFunctions.cpp:
(WTR::isWebPlatformTestURL):
* Tools/TestRunnerShared/WPTFunctions.h:
* Tools/WebKitTestRunner/WebKitTestRunner.xcodeproj/project.pbxproj:
* Tools/WebKitTestRunner/mac/TestControllerMac.mm:
(WTR::shouldEnableAsyncOverflowScrolling):
(WTR::TestController::platformSpecificFeatureDefaultsForTest const):

Canonical link: <a href="https://commits.webkit.org/270071@main">https://commits.webkit.org/270071@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/54d4fa583bcfae81598a0904c605687e9090f49e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/24420 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/2823 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/25623 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/26551 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/22464 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/24691 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/4466 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/169 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/22880 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/24666 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/2036 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/21092 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/27137 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/1773 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/22017 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/28223 "Passed tests") | 
| | [❌ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/22261 "Found 1 new API test failure: TestWebKitAPI.ServiceWorker.WindowClientNavigate (failure)") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/22337 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/26014 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/1724 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/64 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/3009 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/5859 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/2151 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/2097 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

smfr commented on Tools/TestRunnerShared/TestFeatures.cpp:
```diff
@@ -105,9 +105,8 @@ static std::optional<double> overrideDeviceScaleFactorForTest(const std::string&
 }
 
-static bool shouldDumpJSConsoleLogInStdErr(const std::string& pathOrURL)
+static bool isValidWPTURL(const URL& url)
```
What does it mean to be a "valid" WPT url?

aprotyas replied:
Oh I'm mainly trying to say this is both a valid `URL` and is running on one of the ports where WPTs are expected to run, so for the lack of a better term went with `isValidWPTURL`. Thoughts?

smfr replied:
`isWPTTestURL` perhaps.

nt1m replied:
Test is redundant with the acronym WPT

aprotyas replied:
Renamed to `isWebPlatformTestURL` in https://github.com/WebKit/WebKit/commit/e1b21d5218940d04b200392cc916d0c453f8d3de.

aprotyas committed:
[macOS WKTR] Enable async overflow scrolling by default for WPTs
https://bugs.webkit.org/show_bug.cgi?id=263810
rdar://problem/117609234

Reviewed by Tim Nguyen.

This commit enables async overflow scrolling by default for WPTs on
macOS by hardcoding enablement of the `AsyncOverflowScrollingEnabled` web
preference within `TestControllerMac` when the test runner is running
WPTs.

The heuristic for determining whether the test running is a WPT is
simply the fact that the test path is a URL and is on the ports/hosts
specified in imported/w3c/resources/config.json, as prior art from
`shouldDumpJSConsoleLogInStdErr` shows.

We keep the concept of a valid WPT URL in a separate
`isWebPlatformTestURL` helper method in WPTFunctions.h that both
`shouldDumpJSConsoleLogInStdErr` and `shouldEnableAsyncOverflowScrolling`
can consult. To get all the definitions in the right place and satisfy
the linker, this commit also touches some of target membership for WKTR.

* LayoutTests/platform/mac-wk2/TestExpectations:
* Tools/TestRunnerShared/TestFeatures.cpp:
(WTR::shouldDumpJSConsoleLogInStdErr):
* Tools/TestRunnerShared/WPTFunctions.cpp:
(WTR::isWebPlatformTestURL):
* Tools/TestRunnerShared/WPTFunctions.h:
* Tools/WebKitTestRunner/WebKitTestRunner.xcodeproj/project.pbxproj:
* Tools/WebKitTestRunner/mac/TestControllerMac.mm:
(WTR::shouldEnableAsyncOverflowScrolling):
(WTR::TestController::platformSpecificFeatureDefaultsForTest const):

Canonical link: https://commits.webkit.org/270071@main
